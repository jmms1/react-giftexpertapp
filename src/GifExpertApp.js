import React, { Fragment, useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Game of Thrones'];

    const [categories, setCategories] = useState(['Game of Thrones']);

    // const handleAdd = (  ) => {
    //     // setCategories( ...categories, 'HunterXHunter' );
    //     setCategories( cats => [ ...cats, ] );

    // }

    return(
    <Fragment>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr />



        <ol>
            {
             categories.map( category =>
                 <GifGrid
                 key= {category}
                 category = {category}/>
             )}
        </ol>

    </Fragment>

    )
}


export default GifExpertApp;