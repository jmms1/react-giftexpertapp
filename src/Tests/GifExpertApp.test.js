import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test en GifExpertApp', () => {
    
    const wrapper = shallow(<GifExpertApp />)

    test('render correcto ', () => {
        
        expect(wrapper).toMatchSnapshot();
    })
    



})
