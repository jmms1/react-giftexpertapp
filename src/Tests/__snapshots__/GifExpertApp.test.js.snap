// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en GifExpertApp render correcto  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Game of Thrones"
      key="Game of Thrones"
    />
  </ol>
</Fragment>
`;
